### PR TITLE
Add iOS and tvOS CMake GitHub Actions checks 

### PR DESCRIPTION
**Pull request by Ravbug**
Adds CMake builds to `main.yml` for iOS and tvOS. I reported #9462 because I was unable to use these configurations. 

## Description
Adds these builds as entries to the matrix in `main.yml`. Doesn't add framework builds, only GNU prefix builds.

### CODE REVIEW

**Comment by madebr** [comment]
I was thinking about folding `ios.yml` into `main.yml`, but since the xcode projects are often temporarily broken, I opted out.

**Comment by madebr** [comment]
@Ravbug 
Can you look into what testgles needs to satisfy the missing symbols?
Can you also look why tvos is failing? `SDL_PLATFORM_IOS` should not be defined there (we have `SDL_PLATFORM_TVOS`).
And finaly, can you also verify my watchOS changes?

**Comment by Semphriss** [comment]
Just reporting here what I commented on #9462, I'd like to give a look at the dialog thing to see if I missed something. Not sure if this PR comes packed with fixes for that already, but I wanted to say that I'm looking into it.

**Comment by Ravbug** [comment]
@madebr 
>  Can you look into what testgles needs to satisfy the missing symbols? Can you also look why tvos is failing?

I believe it needs `OpenGLES.framework`. Both are failing at the same spot. 

>  And finaly, can you also verify my watchOS changes?

watchOS will need some additional effort to support (we probably don't want to do that in this PR). I configured with `cmake -G "Xcode" -DCMAKE_SYSTEM_NAME=watchOS` and I get a couple errors:
```cpp
/Users/admin/Downloads/SDL/src/hidapi/ios/hid.m:56:10: fatal error: QuartzCore/QuartzCore.h' file not found
#include <QuartzCore/QuartzCore.h>
^~~~~~~~~~~~~~~~~~~~~~~~~
```
It looks like the CMake config is enabling joysticks on watchOS, which aren't supported there. The existing Xcode project doesn't have any watchOS targets, and I don't think UIKit works on watchOS.

**Comment by madebr** [comment]
> @madebr
> 
> > Can you look into what testgles needs to satisfy the missing symbols? Can you also look why tvos is failing?
> 
> I believe it needs `OpenGLES.framework`. Both are failing at the same spot.

Does linking to `OpenGL::


> 
> > And finaly, can you also verify my watchOS changes?
> 
> watchOS will need some additional effort to support (we probably don't want to do that in this PR). I configured with `cmake -G "Xcode" -DCMAKE_SYSTEM_NAME=watchOS` and I get a couple errors:
> 
> ```c++
> /Users/admin/Downloads/SDL/src/hidapi/ios/hid.m:56:10: fatal error: QuartzCore/QuartzCore.h' file not found
> #include <QuartzCore/QuartzCore.h>
> ^~~~~~~~~~~~~~~~~~~~~~~~~
> ```
> 
> It looks like the CMake config is enabling joysticks on watchOS, which aren't supported there. The existing Xcode project doesn't have any watchOS targets, and I don't think UIKit works on watchOS.



> @madebr
> 
> > Can you look into what testgles needs to satisfy the missing symbols? Can you also look why tvos is failing?
> 
> I believe it needs `OpenGLES.framework`. Both are failing at the same spot.

Can you apply this change? Then we won't need multiple ci attempts to get it into green.

> 
> > And finaly, can you also verify my watchOS changes?
> 
> watchOS will need some additional effort to support (we probably don't want to do that in this PR). I configured with `cmake -G "Xcode" -DCMAKE_SYSTEM_NAME=watchOS` and I get a couple errors:
> 
> ```c++
> /Users/admin/Downloads/SDL/src/hidapi/ios/hid.m:56:10: fatal error: QuartzCore/QuartzCore.h' file not found
> #include <QuartzCore/QuartzCore.h>
> ^~~~~~~~~~~~~~~~~~~~~~~~~
> ```
> 
> It looks like the CMake config is enabling joysticks on watchOS, which aren't supported there. The existing Xcode project doesn't have any watchOS targets, and I don't think UIKit works on watchOS.

Let's remove my watchos changes then. Does this mean SDL's video system does not work there?

**Comment by slouken** [comment]
Actually I believe the IOS symbol is defined for TVOS, by design.

**Comment by madebr** [comment]
> Actually I believe the IOS symbol is defined for TVOS, by design.

Thanks. That means `testgles.c` needs:
```c
#if defined(SDL_PLATFORM_IOS) && !defined(SDL_PLATFORM_TVOS)
```

**Comment by Ravbug** [comment]
@madebr 
>> I believe it needs OpenGLES.framework. Both are failing at the same spot.

>Can you apply this change? Then we won't need multiple ci attempts to get it into green.

https://github.com/libsdl-org/SDL/blob/48c9b1c5316efec842bd0469ce9262c74aaf6a65/test/CMakeLists.txt#L358-L360
I added linking to GLES here. It looks like a different test is now erroneously linking to Cocoa, which is not present on iOS or tvOS.

**Comment by Ravbug** [comment]
Now it's failing with this message:
```cpp
error: 'glClearColor' is deprecated: first deprecated in iOS 12.0 - OpenGLES API deprecated. (Define GLES_SILENCE_DEPRECATION to silence these warnings) [-Werror,-Wdeprecated-declarations]
    glClearColor(0.0, 0.0, 0.0, 1.0);
```
Do we want to add the `GLES_SILENCE_DEPRECATION` define, or disable this test on iOS and tvOS?

**Comment by Ravbug** [comment]
Looks like it's now failing because we need to pass in `-DCMAKE_SYSTEM_NAME=`  here:
https://github.com/libsdl-org/SDL/blob/ea9e592b758133bd66a63514d9f29bb515659c60/.github/workflows/main.yml#L140-L148

Do we want to add another field to the matrix?

**Comment by Ravbug** [comment]
Looks like it's passing now. Are we good to merge? @slouken

File: cmake/sdlplatform.cmake [hunk 1]
```diff
@@ -6,17 +6,18 @@ macro(SDL_DetectCMakePlatform)
...
-      set(SDL_CMAKE_PLATFORM Darwin)
-    elseif(CMAKE_SYSTEM_NAME MATCHES ".*MacOS.*")
-      set(SDL_CMAKE_PLATFORM MacosX)
+    if(CMAKE_SYSTEM_NAME MATCHES ".*(Darwin|MacOS).*")
+      set(SDL_CMAKE_PLATFORM macOS)
     elseif(CMAKE_SYSTEM_NAME MATCHES ".*tvOS.*")
       set(SDL_CMAKE_PLATFORM tvOS)
     elseif(CMAKE_SYSTEM_NAME MATCHES ".*iOS.*")
+      set(SDL_CMAKE_PLATFORM watchOS)
+    elseif(CMAKE_SYSTEM_NAME MATCHES ".*watchOS.*")
```
[the remark this answers is not in the document]
Comment (madebr) [reply cmake/sdlplatform.cmake:15]: No you aren't. I didn't see this in the diff. You really have eagle eyes :)

[the remark this answers is not in the document]
Comment (madebr) [reply cmake/sdlplatform.cmake:15]: Fixed in 9f83647ae

File: cmake/sdlplatform.cmake [hunk 2]
```diff
@@ -6,17 +6,18 @@ macro(SDL_DetectCMakePlatform)
...
       set(SDL_CMAKE_PLATFORM iOS)
     elseif (CMAKE_SYSTEM_NAME MATCHES "visionOS")
       set(SDL_CMAKE_PLATFORM visionOS)
-      set(VISIONOS ON)       # CMAKE does not set this automatically yet
```
Comment (sezero) [inline cmake/sdlplatform.cmake:19]: Does minimum cmake version we require set this automatically now?

Comment (madebr) [reply cmake/sdlplatform.cmake:19]: No, at the bottom of this function, our cmake script uppercases the value of `SDL_CMAKE_PLATFORM`, and set the variable to a true-ish value.

Comment (sezero) [reply cmake/sdlplatform.cmake:19]: OK

**Pull request merged**